Push for opinions

joesoftware opened the pull request.

joesoftware committed:
Push for opinions

akainic commented on _health-care/_js/_components/gender-input.jsx:
```diff
@@ -0,0 +1,58 @@
+    if (field === '') {
+      return true;
+    }
+    return /^[a-zA-Z '\-]+$/.test(field);
```
I don't think you need this regex check because the field is a dropdown, so we're restricting what the user can input anyway

joesoftware replied:
Not that it's any better, but I put an explicit check for " ", "F", and "M".

akainic commented on _health-care/_js/_components/gender-input.jsx:
```diff
@@ -0,0 +1,58 @@
+  }
+
+  validate(field) {
+    if (field === '') {
```
This raises a good question. I don't have a sense right now of what fields are required or not. I don't know if we can assume that only fields with the "Required" span text to it are required. It seems to me that gender should be a required field. But if it's not required, I don't think we should have to run validations at all.

akainic commented on _health-care/_js/_components/name-and-general-info-section.jsx:
```diff
@@ -24,6 +25,8 @@ class NameAndGeneralInfoSection extends React.Component {
                   onUserInput={(update) => {this.props.onStateChange('mothersMaidenName', update);}}/>
               <SocialSecurityNumber ssn={this.props.data.socialSecurityNumber}
                   onUserInput={(update) => {this.props.onStateChange('socialSecurityNumber', update);}}/>
+              <GenderInput name={this.props.data.gender}
```
The prop should be `gender`, not `name`.

akainic replied:
Also, `this.props.data.gender` is referencing the `applicationData` specified in `health-care-app.jsx`. I'm assuming if gender isn't required we don't need to set it in `applicationData`? And this initially will just be undefined? Perhaps @awong-dev can weigh in here.

akainic commented on _health-care/_js/_components/gender-input.jsx:
```diff
@@ -0,0 +1,58 @@
+          <select
+              id={`${this.id}-gender`}
+              ref="gender"
+              type="text"
```
Don't need to specify `type="text"` for a select.

akainic commented on _health-care/_js/_components/gender-input.jsx:
```diff
@@ -0,0 +1,58 @@
+              type="text"
+              value={this.props.gender}
+              onChange={this.handleChange}>
+              <option value="M">Male</option>
```
Let's add a default blank option before male! With a value of `0`, I see that's what we're using elsewhere. And then when we have more information about this field we can change it later.

akainic commented on _health-care/_js/_components/gender-input.jsx:
```diff
@@ -0,0 +1,58 @@
+  handleChange() {
+    const gender = this.refs.gender.value;
+
+    if (this.validate(gender)) {
```
Nice, you switched this conditional, makes more sense this way :)

akainic commented:
Looks like you have some lint errors? Might be why the build failed. Try fixing those and see what happens! Also, you created a React component, yay! Passing to @awong-dev for a final review.

joesoftware committed:
Fix lint issue.
Make gender default to blank (" ").

joesoftware committed:
Made changes suggested by Anne

joesoftware committed:
Rename Gender component.
Add AddressState component.

akainic commented on _health-care/_js/_components/address-state.jsx:
```diff
@@ -0,0 +1,112 @@
+  }
+
+  validate(field) {
+    if (field === '') return true;
```
This is not the right syntax. It should be: 
```
if (field === '') {
    return true;
}
```
Also, we're returning true either way. Is this a required field? If not, I don't think we even need to validate at all. If you want to keep the validation logic in here once we find out more information about what fields are required or not, I would just include `return true;` for now without checking `if (field === '')`.

joesoftware replied:
Removing the `validate` function completely in the next push.

akainic commented on _health-care/_js/_components/address-state.jsx:
```diff
@@ -0,0 +1,112 @@
+  }
+
+  handleChange() {
+    const state = this.refs.StateInput.value;
```
Do you mean just `this.refs.state.value` here?

joesoftware replied:
Yes, this is a bug. Not sure why it didn't fail, but I just noticed that using `this.refs.StateInput.value` kept the form from persisting the field value. Will correct in next push.

joesoftware committed:
Renamed gender component.
Removed validate functions from Gender and AddressState components.
Corrected AddressState's handleChange function to use correct name.

awong-dev commented on _health-care/_js/_components/address-state.jsx:
```diff
@@ -0,0 +1,107 @@
+              ref="state"
+              value={this.props.state}
+              onChange={this.handleChange}>
+            <option value="0"></option>
```
Use "" instead of "0".

akainic commented on _health-care/_js/_components/gender.jsx:
```diff
@@ -0,0 +1,50 @@
+  handleChange() {
+    const gender = this.refs.gender.value;
+
+    if (this.validate(gender)) {
```
This has to be removed as well.

awong-dev commented on _health-care/_js/_components/address-state.jsx:
```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import _ from 'lodash';
+
+class AddressState extends React.Component {
```
Needs a class comment explaining the basic API of this component.

akainic commented on _health-care/_js/_components/address-state.jsx:
```diff
@@ -0,0 +1,107 @@
+  handleChange() {
+    const state = this.refs.state.value;
+
+    if (this.validate(state)) {
```
There's no `validate` function now! This if/else conditional should be removed.

awong-dev commented on _health-care/_js/_components/gender.jsx:
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import _ from 'lodash';
+
+class Gender extends React.Component {
```
Also needs class component.

awong-dev commented on _health-care/_js/_components/gender.jsx:
```diff
@@ -0,0 +1,50 @@
+              ref="gender"
+              value={this.props.gender}
+              onChange={this.handleChange}>
+            <option value="0"></option>
```
"0" => ""

awong-dev commented:
Oops. This was merged w/o LGTM. That's okay this time, but let's try to avoid that going forward.  Also, the PR name + description should be considered part of the code review (@akainic tagging you for FYI here) as they become the commit message of the merge node on mainline.

I added a few comments. Please address in a followup PR.

awong-dev commented:
We are also missing unittests for these components.
